bip-taro-vm.mediawiki

```diff
@@ -109,6 +130,37 @@ follows:
 ## Let the output script be the first 32-bytes of an MS-SMT tree with a single element of the serialized TLV leaf of the collectible. 
 ### The key for this single value is <code>sha256(asset_key_family || asset_id || asset_script_key)</code>. If a <code>asset_key_family</code> field isn't specified, then 32-bytes of zeroes should be used in place.
 
+The following algorithm implements the output mapping required for full state
+transition verification:
+<source lang="python">
+make_virtual_txout(leaf: TaroLeaf) -> (MerkleSumRoot, TxOut):
+    match leaf.asset_type:
+        case Normal:
+            tx_out = NewTxOut(
+                pk_script=[OP_1 OP_DATA_32 leaf.split_commitment_root.hash], 
+                value=leaf.split_commitment_root.sum_value,
+            )
+
+            return leaf.split_commitment_root, tx_out
+
+        case Collectible:
+            output_smt = new_ms_smt()
+            output_smt.insert(
+                key=sha256(leaf.asset_key_family || leaf.asset_id || leaf.asset_script_key)
+                value=leaf.serialize_tlv(),
+                sum_value=1,
+            )
```

Comment:
That's covered above in the existing case? 

Or you mean that right now, it isn't 100% specifid if a split commitment _always_ needs to be in place even if it's spending an entire UTXO (your case above)? If we do this, then we might also want to require the same for collectible transfers, as that'll make things more uniform (you just always place the root here vs making this tree with a single leaf (same thing really)).

Reply:
> In this case, the commitment root would just be the root of the empty tree then (since no splits happen and nothing gets committed to the split tree)?

So we'd either not include it, or if we did, then it would be _just_ that leaf. Today if I make a split and it has a single external split/change-output, then two items are in the tree: my main split and the child split. Both of these need to add up to the input sum, or the theoretical input split amount.